### ISSUE TITLE
Win Arm64 and Arm32 crossgen fails due to JIT asserts

### ISSUE THREAD

**Issue by gkhanna79** [dotnet/coreclr#8989]
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=525330&tab=details (arm)
 
```
Assert failure(PID 6352 [0x000018d0], Thread: 7604 [0x1db4]): Assertion failed '(block->bbFlags & BBF_FINALLY_TARGET) != 0' in 'System.Reflection.LoaderAllocatorScout:Finalize():this' (IL size 63)
 
    File: e:\a\_work\35\s\src\jit\flowgraph.cpp Line: 11661
    Image: E:\A\_work\35\s\bin\Product\Windows_NT.arm.Checked\x86\crossgen.exe
```
 
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=525329&tab=details (arm64)
 
```
Assert failure(PID 780 [0x0000030c], Thread: 5808 [0x16b0]): Assertion failed '(lastConsumedNode == nullptr) || (node->gtUseNum == -1) || (node->gtUseNum > lastConsumedNode->gtUseNum)' in 'System.Globalization.HebrewCalendar:GetDatePart(long,int):int:this' (IL size 467)
 
    File: e:\a\_work\12\s\src\jit\codegenlinear.cpp Line: 1101
    Image: E:\A\_work\12\s\bin\Product\Windows_NT.arm64.Checked\x64\crossgen.exe
```

Also reported in today's (1/18) build - https://devdiv.visualstudio.com/DevDiv/_build?buildId=526476.

**Comment by CarolEidt** [comment]
The second assert (arm64) has been fixed with #8991. @jashook is looking at the arm one.

**Comment by gkhanna79** [comment]
@CarolEidt Thank you. Today's build confirms the issue is addressed and the only outstanding issue is that for WinArm32. @jashook How is that looking since this build break is preventing publishing of nuget packages from the official builds?

**Comment by jashook** [comment]
#9013 should fix the crossgen assertion, but it does not address the root cause. Will open a new issue and assign it to Andy after #9013 is in.

**Comment by jashook** [comment]
Build break is fixed with #9013. #9015 is opened to address the root issue. /cc @AndyAyersMS
